src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -489,10 +488,6 @@ private TypeSymbol GetTypeOfTypeDef(TypeDefinitionHandle typeDef, out bool isNoP
                     {
                         isNoPiaLocalType = true;
                     }
-                    else
-                    {
-                        isNoPiaLocalType = false;
```

Comment:
Consider removing assignment from beginning of the method instead, to ensure all exit points explicitly set the out parameter.